test/e2e/metering_manual_install_tests.go

```diff
@@ -107,6 +108,141 @@ func testManualMeteringInstall(
+				},
+			},
+		}
+		_, err = ctx.Client.CoreV1().Namespaces().Create(context.Background(), n, metav1.CreateOptions{})
+		if err != nil {
+			return err
+		}
+		ctx.Logger.Debugf("Created the %s namespace", ctx.Namespace)
+	}
+
+	//decode and create yamls except for persistentvolume
+
+	type nfsManifestItem struct {
+		resourceType interface{}
+		resourceFile string
+		actions      []string
+	}
+	type nfsServer struct {
+		CRD *apiextv1.CustomResourceDefinition
+	}
+
+	type nfsService struct {
+		CRD *apiextv1.CustomResourceDefinition
+	}
+
+	type nfsStorageclass struct {
+		CRD *apiextv1.CustomResourceDefinition
+	}
+	var nfsManifestItems = map[string]nfsManifestItem{
+		"nfsServer":
+		{
+
+			resourceType: nfsServer{},
+			resourceFile: "/test/testhelpers/nfs_server.yaml",
+			actions:      []string{"CREATE"},
+		},
+		"nfsService":
+		{
+			resourceType: nfsService{},
+			resourceFile: "/test/testhelpers/nfs_service.yaml",
+			actions:      []string{"CREATE", "GET"},
+		},
+		"nfsStorageclass":
+		{
+			resourceType: nfsStorageclass{},
+			resourceFile: "/test/testhelpers/nfs_storageclass.yaml",
+			actions:      []string{"CREATE"},
+		},
+	}
+
+	for _, manifest := range nfsManifestItems {
+
+		err := deploy.DecodeYAMLManifestToObject(nfsManifestItems["nfsServer"].resourceFile, nfsManifestItems["nfsServer"].resourceType)
+		err := deploy.DecodeYAMLManifestToObject(nfsManifestItems["nfsService"].resourceFile, nfsManifestItems["nfsService"].resourceType)
+		err := deploy.DecodeYAMLManifestToObject(nfsManifestItems["nfsStorageclass"].resourceFile, nfsManifestItems["nfsStorageclass"].resourceType)
+		if err != nil {
+			return fmt.Errorf("failed to decode the %s YAML service manifest: %v", err, manifest.resourceFile)
+		}
+
+		for _, action := range manifest.actions {
+			if action == "CREATE" {
+				ctx.Client.CoreV1().Namespaces().Create(context.Background(), &nfsManifestItems[0].resourceType, metav1.CreateOptions{})
+			}
+			ctx.Logger.Debugf("Created the %s nfs_server %s nfs_service %s nfs_storageclass", nfsManifestItems["nfsServer"],nfsManifestItems["nfsService"].resourceFile, nfsManifestItems["nfsStorageclass"].resourceFile)
+			if action == "GET" {
+
+				foo, err := ctx.Client.CoreV1().Services(ctx.Namespace).Get(context.Background(), "nfs-service", metav1.GetOptions{})
+				if err != nil && !apierrors.IsNotFound(err) {
+					return nil
+				}
+			}
+		}
+	}
+
+	//  - Wait for service to populate spec.ClusterIP (poll.Wait functionality)
+	// poll has a closure that polls for the server IP to get populated
+	err := wait.Poll(3*time.Second, 5*time.Minute, func() (done bool, err error) {
+
+		pvc, err := ctx.Client.CoreV1().Services(ctx.Namespace).Get(context.TODO(), "nfs-service", metav1.GetOptions{})
+		// if theres an error don't poll
+		if err != nil {
+			return false, err
+		}
+		// clusterIP is populated, stop polling
+		return pvc.Spec.nfs.server != "", nil
+	})
+	// no service exists, stop polling
```

Comment:
This comment is misleading, but I would remove it anyways as it doesn't serve much purpose when reading the code.